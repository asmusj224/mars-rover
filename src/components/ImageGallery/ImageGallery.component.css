.image-gallery {
  max-width: 1200px;
  margin: 0 auto;
  margin-top: 2rem; }
  .image-gallery__list {
    list-style: none;
    display: flex;
    flex-wrap: wrap; }
  .image-gallery__item {
    display: flex;
    flex-direction: column;
    width: 250px;
    margin: 0 1rem 1rem 1rem;
    transition: all 0.3s; }
    .image-gallery__item:hover {
      transform: scale(1.1); }
  .image-gallery__image {
    max-width: 100%;
    display: block;
    opacity: 0.7;
    height: 250px;
    border-radius: 3px; }
    .image-gallery__image:hover {
      opacity: 1; }
  .image-gallery__description {
    text-align: center;
    font-size: 14px;
    color: #a9a9a9; }
