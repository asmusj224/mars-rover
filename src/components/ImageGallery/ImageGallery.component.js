import React from 'react';
import './ImageGallery.component.css';

const ImageGallery = ({ roverImages = [] }) => {
    roverImages = roverImages.map(roverImage => {
        return (<li className="image-gallery__item" key={roverImage.id}>
            <img className="image-gallery__image" src={roverImage.img_src} alt={roverImage.rover.name} />
            <span className="image-gallery__description">{roverImage.rover.name} - {roverImage.camera.full_name}</span>
        </li>)
    });


    return (
        <div className="image-gallery">
            <ul className="image-gallery__list">{roverImages}</ul>
        </div>
    );
}

export default ImageGallery;