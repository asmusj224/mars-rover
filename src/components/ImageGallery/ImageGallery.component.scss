@import 'styles/_colors.scss';
.image-gallery {
    max-width: 1200px;
    margin: 0 auto;
    margin-top: 2rem;
    &__list {
        list-style: none;
        display: flex;
        flex-wrap: wrap;
    }
    &__item {
        display: flex;
        flex-direction: column;
        width: 250px;
        margin: 0 1rem 1rem 1rem;
        transition: all 0.3s;
        &:hover {
            transform: scale(1.1);
        }
    }
    &__image {
        max-width: 100%;
        display: block;
        opacity: 0.7;
        height: 250px;
        border-radius: 3px;
        &:hover {
            opacity: 1;
        }
    }
    &__description {
        text-align: center;
        font-size: 14px;
        color: $darkgray;
    }
}