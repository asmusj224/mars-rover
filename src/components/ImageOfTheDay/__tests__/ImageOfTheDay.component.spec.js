import React from 'react';
import renderer from 'react-test-renderer';

import ImageGallery from '../ImageOfTheDay.component';

describe('<ImageGallery />', () => {
    it('renders correctly', () => {
        const component = renderer.create(
            <ImageGallery
                roverImage='roverImage'
                roverName='roverName'
                roverCameraName='roverCameraName' />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

