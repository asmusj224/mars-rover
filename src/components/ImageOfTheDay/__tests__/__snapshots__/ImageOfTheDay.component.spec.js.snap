// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageGallery /> renders correctly 1`] = `
<div
  className="image-of-the-day"
>
  <img
    alt="roverName"
    className="image-of-the-day__image"
    src="roverImage"
  />
  <p
    className="image-of-the-day__rover"
  >
    roverName
     - 
    roverCameraName
  </p>
</div>
`;
